File: lotty-web/src/lib/types/lottery.ts
import type {Participant} from '$lib/types/participant'
import type {Drawing} from '$lib/types/drawing'

export class Lottery {
    name: String
    id: String
    participants: Array<Participant>
    drawResults: Array<Drawing>


    constructor(name: String, id: String, participants: Array<Participant>, drawResults: Array<Drawing>) {
        this.name = name;
        this.id = id;
        this.participants = participants;
        this.drawResults = drawResults;
    }
}

File: lotty-core/src/main/java/dev/myrold/api/Lottery.java
package dev.myrold.api;

import java.util.List;
import java.util.Set;

import io.micronaut.core.annotation.Introspected;

@Introspected
public record Lottery(
    String id,
    String name,
    Set<Participant> participants,
    List<Drawing> drawings
) implements TsidAware {
}
